@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -403,45 +376,84 @@ export class MapControls extends THREE.EventDispatcher {
      * @param deltaAltitude Delta altitude in degrees.
      */
     orbitFocusPoint(deltaAzimuth: number, deltaAltitude: number) {
-        if (this.mapView.projection.type === geoUtils.ProjectionType.Spherical) {
-            return;
-        }
         if (this.inertiaEnabled && this.m_zoomIsAnimated) {
             this.stopZoom();
         }
 
-        this.mapView.camera.getWorldDirection(this.m_currentViewDirection);
-        const currentAzimuthAltitude = utils.directionToAzimuthAltitude(
-            this.m_currentViewDirection
-        );
+        const pitchRad = MapViewUtils.extractYawPitchRoll(this.camera, this.mapView.projection.type)
+            .pitch;
+
+        // In sphere view, from afar, when the horizon is below the screen center (NDC of 0
+        // vertically), find the uppermost NDC so that the interaction does not break.
+        let verticalNDCToRaycast = 0;
+        if (this.mapView.projection.type === geoUtils.ProjectionType.Spherical) {
```

Comment:
Not sure if this is the most correct behaviour from the interaction point of view... 

Since earth at far distances don't cover the entire vFov, tilting from far away will not keep the tilt point in the center of the screen, and when it does, it might mean that that point fall into earth's horizon and is no longer visible.

This also gets worse whenever we introduce any of the acceleration-based algorithms into play, since the rotation in which earth will end up is not deterministic and can translate into inconsistent behaviour (specially when you use the `3D` mapUi button).

I'd stick to limiting the tilt based on the distance to the earth's surface, but that's just my opinionated view on the topic :)

Reply:
If there is a target point in the center of the screen, then this method will not get the earth outside the center of the screen. This math is rather to bring it back.

This is to address the problem of very small zoom levels. The tilt uses to be enabled at up to 45 by default. Problem, if we tilt with the middle mouse button (camera tilt instead of target tilt), then the earth goes out of sight inevitably. Bringing it back with the middle mouse button is a possibility, but this code also allows to bring it back with right mouse button and the "3D" button, and also with the 3 fingers interaction on mobile. Since it's already there, I think it is a strong benefit?

I don't understand the part about acceleration-based algorithms?

Reply:
Update: after discussions with @ninok, the controls should prevent that the center of the screen points in the void (in sphere projection) at all.